### PR TITLE
Support `np.uint16` images

### PR DESCRIPTION
**Pull request by williamshen-nz**
Mostly so we can save the depth images from the RealSense which are in unsigned int16 grayscale.

### CODE REVIEW

**Comment by geyang** [comment]
looking at this now

**Comment by williamshen-nz** [comment]
Simple script to test it works, essentially we need to provide `np.uint16` as the dtype to avoid any normalization, etc. happening to the depth image.

```
import glob
import os

import cv2
import numpy as np

if __name__ == "__main__":
    from ml_logger import logger

    logger.configure(prefix="/william/")
    logger.start()

    root_dir = "2022-10-25_14-30-40"
    rgb_dir = "source"
    depth_dir = "depth"

    rgb_files = glob.glob(os.path.join(root_dir, rgb_dir, "*.png"))
    depth_files = glob.glob(os.path.join(root_dir, depth_dir, "*.png"))

    # Sort by name
    rgb_files.sort()
    depth_files.sort()

    for rgb_path, depth_path in zip(rgb_files, depth_files):
        # Load rgb and convert from BGR to RGB
        rgb = cv2.imread(rgb_path)
        rgb = cv2.cvtColor(rgb, cv2.COLOR_BGR2RGB)
        depth = cv2.imread(depth_path, cv2.IMREAD_UNCHANGED)
        with logger.Sync():
            logger.save_image(rgb, rgb_path)
            logger.save_image(depth, depth_path, dtype=np.uint16)
        print(".", end="")
```


RGB images saved as usual

```
> pwd
/tmp/logging-server/william/2022-10-25_14-30-40/source
> file image-000.png
image-000.png: PNG image data, 1280 x 720, 8-bit/color RGB, non-interlaced
```

Depth is saved as uint16

```
> pwd
/tmp/logging-server/william/2022-10-25_14-30-40/depth
> file image-000.png
image-000.png: PNG image data, 1280 x 720, 16-bit grayscale, non-interlaced
```

File: ml_logger/ml_logger.py [hunk 1]
```diff
@@ -1578,14 +1594,14 @@ def save_images(self, stack, key, n_rows=None, n_cols=None, cmap=None, normalize
         self.client.send_image(key=pJoin(self.prefix, key), data=composite)
         return key
 
-    def save_image(self, image, key: str, cmap=None, normalize=None):
+    def save_image(self, image, key: str, cmap=None, normalize=None, dtype=np.uint8):
```
[the remark this answers is not in the document]
Comment (geyang) [reply ml_logger/ml_logger.py:1597]: can't we just use whatever the image started with, and default to uint8 for normal images?

[the remark this answers is not in the document]
Comment (williamshen-nz) [reply ml_logger/ml_logger.py:1597]: I assumed this was so we could handle images of arbitrary type...

**Pull request merged**